src/api.v2/controllers/experimentController.js

```diff
@@ -129,11 +138,45 @@ const downloadData = async (req, res) => {
   logger.log(`Providing download link for download ${downloadType} for experiment ${experimentId}`);
 
   const downloadLink = await new Experiment().getDownloadLink(experimentId, downloadType);
+
+  logger.log(`Finished providing download link for download ${downloadType} for experiment ${experimentId}`);
   res.json(downloadLink);
 };
 
+
+const cloneExperiment = async (req, res) => {
+  const getAllSampleIds = async (experimentId) => {
+    const { samplesOrder } = await new Experiment().findById(experimentId).first();
+    return samplesOrder;
+  };
+
+  const {
+    params: { experimentId: fromExperimentId, toExperimentId },
+    body: { samplesSubsetIds = await getAllSampleIds(fromExperimentId) },
+  } = req;
+
+  logger.log(`Cloning experiment ${fromExperimentId} into ${toExperimentId}`);
+
+  // toExperiment might be an old experiment with metadata tracks,
```

Comment:
I think this is kind of risky, like reusing old experiment also not sure how you could "select" an old experiment in the UI. In my opinion we should always create a new copy from scratch when you are cloning an experiment. I don't see any benefit in reusing old experiments, but potentially I see many issues (if we don't clean it up correctly)

Reply:
Same as https://github.com/hms-dbmi-cellenics/api/pull/377#discussion_r904927722